pkg/build/controller/strategy/util.go

```diff
@@ -139,6 +142,22 @@ func setupSourceSecrets(pod *kapi.Pod, sourceSecret *kapi.LocalObjectReference)
 	}...)
 }
 
+// setupSecretBuildSources mounts the secrets referenced by the SecretBuildSource
+// into a builder container. It also sets an environment variable that contains
+// a name of the secret and the destination directory.
+func setupSecretBuildSources(pod *kapi.Pod, secrets []buildapi.SecretBuildSource) {
+	for _, s := range secrets {
+		mountSecretVolume(pod, s.Secret.Name, filepath.Join(SecretBuildSourceBaseMountPath, s.Secret.Name), "build")
+		pod.Spec.Containers[0].Env = append(pod.Spec.Containers[0].Env, []kapi.EnvVar{
+			{
+				Name:  fmt.Sprintf("BUILD_SECRET_%s", strings.ToUpper(s.Secret.Name)),
```

Comment:
I ran into issues with this because secretNames can contain characters that are not valid as env var keys (specifically i think "-" was an issue).  you may need to test that and revisit this.